dm/templates/stackdriver_notification_channels/stackdriver_notification_channels.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2019 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" This template creates a Stackdriver Metric Descriptor. """
+
+DEFAULT_MIME_TYPE = "text/markdown"
+
+
+def get_condition_threshold(condition_properties):
+    condition_threshold = {}
+    properties = [
+        'filter',
+        'comparison',
+        'duration',
+        'thresholdValue',
+        'trigger',
+        'aggregations'
+    ]
+    for prop in condition_properties:
+        if prop in properties:
+            condition_threshold[prop] = condition_properties[prop]
+    return condition_threshold
+
+
+def get_policy_conditions(policy_conditions):
+    conditions = []
+    for condition in policy_conditions:
+        conditions.append({
+            'displayName': condition['displayName'],
+            'conditionThreshold': get_condition_threshold(condition)
+        })
+    return conditions
+
+
+def generate_config(context):
+    """ Entry point for the deployment resources. """
+
+    resources = []
+    outputs = []
+    properties = context.properties
+    resource_name = context.env['name']
+    project_id = properties.get('project', context.env['project'])
+    notification_channels = properties["notificationChannels"]
+    policies = properties["policies"]
+
+    notification_channel_names = []
+
+    for index, channel in enumerate(notification_channels):
+        channel_type = channel['typeName']
+        notification_channel_name = '{}-{}-{}'.format(resource_name, channel_type, index)
+        notification_channel_names.append('$(ref.{}.name)'.format(notification_channel_name))
+        resources.append({
+            'name': notification_channel_name,
+            'type': 'gcp-types/monitoring-v3:projects.notificationChannels',
+            'properties': {
+                'name': 'projects/{}'.format(project_id),
+                'type': channel_type,
+                'enabled': channel.get('enabled', True),
```

Comment:
`enabled` is not mentioned in Schema.
Also default values should be set in schema, not deep in code. ( Same for policy)